### PR TITLE
Handle empty strings in type coercion.

### PR DESCRIPTION
**Pull request by hidmic**
Follow-up after #438.

### CODE REVIEW

File: launch/launch/utilities/type_utils.py [hunk 1]
```diff
@@ -231,6 +231,10 @@ def coerce_to_type(
     :return: `value` coerced to `data_type`.
     """
     def convert_as_yaml(value, error_msg):
+        # Forward empty strings
+        if not value:
```
Comment (dirk-thomas) [inline launch/launch/utilities/type_utils.py:235]: I wondering if this need to match what the comment suggests. This condition will be true for various other kind of input, like `0`, `None`, an empty list/set/dict?

**Comment by hidmic** [comment]
CI up to `launch` and  `test_launch_ros`:

* Linux [Build Status](https://ci.ros2.org/job/ci_linux/11703/)
* Linux aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/6800/)
* MacOS [Build Status](https://ci.ros2.org/job/ci_osx/9554/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/11652/)

File: launch/launch/utilities/type_utils.py (hunk 1 above)
Comment (dirk-thomas) [reply launch/launch/utilities/type_utils.py:235]: Nvm, the function argument is supposed to be a string.

Comment (hidmic) [reply launch/launch/utilities/type_utils.py:235]: If you look at line 251, it is ensured that value is a `str` object.

**Pull request merged**